manifests/params.pp

```diff
@@ -145,22 +151,26 @@
         }
       }
 
-      $service_name       = 'elasticsearch'
-      $service_hasrestart = true
-      $service_hasstatus  = true
-      $service_pattern    = $service_name
-      $defaults_location  = '/etc/sysconfig'
-      $pid_dir            = '/var/run/elasticsearch'
     }
     'Debian', 'Ubuntu': {
       $service_name       = 'elasticsearch'
       $service_hasrestart = true
       $service_hasstatus  = true
       $service_pattern    = $service_name
-      $service_providers  = 'init'
       $defaults_location  = '/etc/default'
-      $init_template      = 'elasticsearch.Debian.erb'
-      $pid_dir            = false
+
+      case $::operatingsystemmajrelease {
+        '8': {
```

Comment:
I think Ubuntu will have to get a separate block so we can make the differences more clearly.
I hate to have those long if statements, especially if they have an 'or' case.
it invites to more horrible stuff :-)